code/src/main/java/frc/robot/commands/OdometryCommand.java

```diff
@@ -0,0 +1,109 @@
+import frc.robot.common.*;
+
+/**
+ * A odometry class that uses an array of points to follow in a sequential order
+ */
+public class OdometryCommand extends TraceableCommand {
+    private final IDrivetrainSubsystem m_subsystem;
+    private final Pose2d[] m_path;
+    private PIDController m_xPIDController;
+    private PIDController m_yPIDController;
+    private ProfiledPIDController m_thetaController;
+    private int currentPosIndex = 0;
+
+    /**
+     * Creates a new OdometryCommand.
+     * 
+     * @param logger    The logger used to log the data
+     * @param subsystem The subsystem used by this command.
+     * @param path      The array of points to get to.
+     * @param config    The RobotConfig file that is used for this command.
+     */
+    @Inject
+    public OdometryCommand(ILogger logger, IDrivetrainSubsystem subsystem, Pose2d[] path, RobotConfig config) {
+        super(logger);
+        m_subsystem = subsystem;
+        m_path = path;
+
+        m_xPIDController = new PIDController(config.Odometry.pX, config.Odometry.iX, config.Odometry.dX);
+        m_yPIDController = new PIDController(config.Odometry.pY, config.Odometry.iY, config.Odometry.dY);
+        m_thetaController = new ProfiledPIDController(config.Odometry.pTheta, config.Odometry.iTheta,
+                config.Odometry.dTheta, new TrapezoidProfile.Constraints(2, 1));
+
+        m_xPIDController.setTolerance(config.Odometry.toleranceX);
+        m_xPIDController.disableContinuousInput();
+        m_xPIDController.reset();
+
+        m_yPIDController.setTolerance(config.Odometry.toleranceY);
+        m_yPIDController.disableContinuousInput();
+        m_yPIDController.reset();
+        // Use addRequirements() here to declare subsystem dependencies.
+        addRequirements(subsystem);
+
+    }
+
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+        super.initialize();
+        m_xPIDController.setSetpoint(m_path[0].getTranslation().getX());
+        m_yPIDController.setSetpoint(m_path[0].getTranslation().getY());
+    }
+
+    // Called every time the scheduler runs while the command is scheduled.
+
+    @Override
+    public void execute() {
+        // Determines if the robot is the current target point.
+        // If true it moves the setpoints to the next position
+        if (m_xPIDController.atSetpoint() && m_yPIDController.atSetpoint() && m_thetaController.atGoal()) {
+            currentPosIndex++;
+
+            if (currentPosIndex < m_path.length) {
+                m_xPIDController.setSetpoint(m_path[currentPosIndex].getTranslation().getX());
+                m_yPIDController.setSetpoint(m_path[currentPosIndex].getTranslation().getY());
+                m_thetaController.setGoal(m_path[currentPosIndex].getRotation().getRadians());
+                m_xPIDController.reset();
+                m_yPIDController.reset();
+
+            }
+
+        }
```

Comment:
I don't understand why this needs to be outside the subsystem. I understand that you need something to keep polling the subsystem for position, however all the calculations seem like they can be done just as well in the subsystem with check for current x,y or simply a amIThereYet method. I'm most concerned that this is going to be hard to mock.

Is one of the reasons due to the amount of code needed in the subsystem? I maybe missing something.